### PR TITLE
WT-10553 Create performance test for ARM

### PR DESCRIPTION
**Pull request by Siddhartha8899**

### CODE REVIEW

File: test/evergreen_develop.yml [hunk 1]
```diff
@@ -71,6 +71,71 @@ buildvariants:
...
+    pip3_binary: '/opt/mongodbtoolchain/v4/bin/pip3'
+    virtualenv_binary: '/opt/mongodbtoolchain/v4/bin/virtualenv'
+    smp_command: -j $(echo "`grep -c ^processor /proc/cpuinfo` * 2" | bc)
+    cmake_generator: Ninja
+    make_command: ninja
+    collection: AllPerfTestsARM64
+  tasks:
+    - name: compile
+    - name: ".btree-perf"
+    # Disable LSM perf tests till the support for LSM is restored.
```
Comment (ajmorton) [inline test/evergreen_develop.yml:98]: I know this comment was present already but could we add a FIXME-WT-8867 to all the comments mentioning LSM being reenabled?

**Review by ajmorton — APPROVED** [review]
LGTM 🧪

File: test/evergreen_develop.yml (hunk 1 above)
Comment (Siddhartha8899) [reply test/evergreen_develop.yml:98]: Updated in [eff9519](https://github.com/wiredtiger/wiredtiger/pull/8909/commits/eff95194e797b614fcb33c3dbea59df5256a9fbf)

**Comment by Siddhartha8899** [comment]
I have tested this PR by running the perf tests. All the tests ran successfully and the results/stats/documents were uploaded to Atlas! 

I will work on creating charts for these tests and share the public dashboard link in the storage channel once the PR is merged.

**Review by lukech — COMMENTED** [review]
Thanks for making the change, Sid. It's worth trying to leverage the yaml template feature to save most of the duplicated lines (between the 2 perf builders). [Here](https://github.com/mongodb/mongo/blob/cd847a70d11509c22d455710a9779328f150b642/etc/evergreen.yml#L157) is an example from mongo evergreen.yml

**Comment by Siddhartha8899** [comment]
> 2e783d4

Updated in [2e783d4](https://github.com/wiredtiger/wiredtiger/pull/8909/commits/2e783d470ac028ac87310996387b38f5d807d024)

File: test/evergreen_develop.yml [hunk 2]
```diff
@@ -3,34 +3,16 @@
...
 - filename: test/evergreen.yml
 
-
 # Build variants that need to be run only on the develop branch.
 buildvariants:
-- name: ubuntu2004-perf-tests
-  display_name: Ubuntu 20.04 Performance tests
+# Template for perf-tasks
+- &perf-tasks-template
+  name: ubuntu2004-perf-tests
```
Comment (lukech) [inline test/evergreen_develop.yml:10]: Is it clearer to include "template" in the name, preferably matching the name in the line above?

File: test/evergreen_develop.yml [hunk 3]
```diff
@@ -71,6 +53,37 @@ buildvariants:
...
+    CC_OPTIMIZE_LEVEL: -DCC_OPTIMIZE_LEVEL=-O3
+    HAVE_DIAGNOSTIC: -DHAVE_DIAGNOSTIC=0
+    CMAKE_TOOLCHAIN_FILE: -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/mongodbtoolchain_stable_gcc.cmake
+    CMAKE_INSTALL_PREFIX: -DCMAKE_INSTALL_PREFIX=$(pwd)/cmake_build/LOCAL_INSTALL
+    python_binary: '/opt/mongodbtoolchain/v4/bin/python3'
+    pip3_binary: '/opt/mongodbtoolchain/v4/bin/pip3'
+    virtualenv_binary: '/opt/mongodbtoolchain/v4/bin/virtualenv'
+    smp_command: -j $(echo "`grep -c ^processor /proc/cpuinfo` * 2" | bc)
+    cmake_generator: Ninja
+    make_command: ninja
```
Comment (lukech) [inline test/evergreen_develop.yml:75]: These settings are common between the 2 builders. Why not move them up into the builder template, and simply quote the expansion template here?

Comment (Siddhartha8899) [reply test/evergreen_develop.yml:75]: There is one setting that is different -  `collection`.

Comment (Siddhartha8899) [reply test/evergreen_develop.yml:75]: My bad, I think I get what you are saying.

File: test/evergreen_develop.yml (hunk 2 above)
Comment (Siddhartha8899) [reply test/evergreen_develop.yml:10]: I tried that, but for some reason it gives me this error when I try to create a patch build.
```
Unexpected reply from server (500 Internal Server Error): 400 (Bad Request): invalid patched project config: invalid patched config syntax: buildvariant 'ubuntu2004-perf-template' must either specify run_on field or have every task specify run_on
buildvariant 'ubuntu2004-perf-template' does not have a display name
```

File: test/evergreen_develop.yml (hunk 3 above)
Comment (lukech) [reply test/evergreen_develop.yml:75]: I think we can exclude the "collection" setting from the expansion template, and explicitly specify it right after quoting the template (like what's implemented in lines 83-85 below).

File: test/evergreen_develop.yml (hunk 2 above)
Comment (lukech) [reply test/evergreen_develop.yml:10]: Let me give a try to see if I can make it work :)

File: test/evergreen_develop.yml (hunk 3 above)
Comment (Siddhartha8899) [reply test/evergreen_develop.yml]: I like that :). Updated in [f77d924](https://github.com/wiredtiger/wiredtiger/pull/8909/commits/f77d924a611b196ad162bb62eb9f463d52d53da2)

Comment (Siddhartha8899) [reply test/evergreen_develop.yml]: Corrected commit updated in [f4f07b1](https://github.com/wiredtiger/wiredtiger/pull/8909/commits/f4f07b108ab02ff0b7ebb6c5697cbbf602537217)

File: test/evergreen_develop.yml [hunk 4]
```diff
@@ -3,15 +3,12 @@
...
 
-
-# Build variants that need to be run only on the develop branch.
-buildvariants:
-- name: ubuntu2004-perf-tests
-  display_name: Ubuntu 20.04 Performance tests
+variables:
+# Template for perf-tasks
+- &perf-tasks-template
+  name: perf-tasks-template
```
Comment (ajmorton) [inline test/evergreen_develop.yml:9]: nit: I saw some discussion earlier that we had to keep name, but now the template has been moved to `variables` is it possible to drop the field?

**Review by ajmorton — APPROVED** [review]
LGTM 🚗

File: test/evergreen_develop.yml (hunk 4 above)
Comment (Siddhartha8899) [reply test/evergreen_develop.yml]: Yes, updated in [14c2725](https://github.com/wiredtiger/wiredtiger/pull/8909/commits/14c27259f64e056ac59eeb36ea4d77b0c9e571cc)

**Review by lukech — APPROVED** [review]
LGTM with a nit.

**Pull request merged**